sql/catalyst/src/main/java/org/apache/spark/sql/util/CaseInsensitiveStringMap.java

```diff
@@ -40,11 +40,16 @@ public static CaseInsensitiveStringMap empty() {
     return new CaseInsensitiveStringMap(new HashMap<>(0));
   }
 
+  private final Map<String, String> original;
+
   private final Map<String, String> delegate;
 
   public CaseInsensitiveStringMap(Map<String, String> originalMap) {
-    this.delegate = new HashMap<>(originalMap.size());
-    putAll(originalMap);
+    original = new HashMap<>(originalMap);
+    delegate = new HashMap<>(originalMap.size());
+    for (Map.Entry<? extends String, ? extends String> entry : originalMap.entrySet()) {
+      delegate.put(toLowerCase(entry.getKey()), entry.getValue());
```

Comment:
If the key is already in the case insensitive map, we should fail and say duplicated keys detected.

Reply:
How about showing warning message here?
E.g.
```
spark.read.option("foo", "bar").option("fOO", "bar")....
```
This is valid.